src/main/java/com/yugabyte/sample/apps/AppBase.java

```diff
@@ -218,12 +218,14 @@ protected Connection getPostgresConnection(String database, String username, Str
         String connectStr = String.format("%s//%s:%d/%s", url, contactPoint.getHost(),
                 contactPoint.getPort(),
                 database);
-        if (!username.equalsIgnoreCase("yugabyte") || !username.equalsIgnoreCase("postgres")){
+        if (!username.equalsIgnoreCase("yugabyte") && !username.equalsIgnoreCase("postgres")){
           Properties newProps = new Properties();
           newProps.setProperty("user", "yugabyte");
+          newProps.setProperty("password", appConfig.ybPassword);
           Connection controlConnection = DriverManager.getConnection(connectStr, newProps);
           Statement st = controlConnection.createStatement();
           String grantPermission = String.format("grant create on schema public to %s with grant option;", username);
+          LOG.info("Granted create permission to user: " + username);
           st.execute(grantPermission);
           controlConnection.close();
```

Comment:
Consider putting this in a `try-finally` block
```suggestion
          Connection controlConnection = null;
          try {
            controlConnection = DriverManager.getConnection(connectStr, newProps);
            Statement st = controlConnection.createStatement();
            String grantPermission = String.format("grant create on schema public to %s with grant option;", username);
            LOG.info("Granted create permission to user: " + username);
            st.execute(grantPermission);
          } finally {
            if (controlConnection != null) {
              controlConnection.close();
            }
          }
```